virtual_ecosystem/core/base_model.py

```diff
@@ -475,6 +596,13 @@ class ExampleModel(
                 defined
             TypeError: If model_name is not a string
         """
+        if cls.__init__ != BaseModel.__init__:
+            raise NotImplementedError("Model subclasses cannot override __init__.")
+
+        if cls.update != BaseModel.update:
+            raise NotImplementedError(
+                "Model subclasses cannot override the update method."
+            )
```

Comment:
Because we're comparing identities, I wonder if this is more correct:

```suggestion
        if cls.__init__ is not BaseModel.__init__:
            raise NotImplementedError("Model subclasses cannot override __init__.")

        if cls.update is not BaseModel.update:
            raise NotImplementedError(
                "Model subclasses cannot override the update method."
            )
```

Reply:
Not entirely sure, to be honest. The `!=` works - I had to change a few models because of that - so I'd leave it as it is.